src/Plugin/RulesAction/EntitySave.php

```diff
@@ -41,13 +42,15 @@ class EntitySave extends RulesActionBase {
   protected $saveLater = TRUE;
 
   /**
-   * {@inheritdoc}
+   * Saves the Entity.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity to be saved.
    */
-  public function execute() {
+  protected function doExecute(EntityInterface $entity) {
     // We only need to do something here if the immediate flag is set, otherwise
     // the entity will be auto-saved after the execution.
     if ((bool) $this->getContextValue('immediate')) {
```

Comment:
The "immediate" context should also come in as method parameter, same as the entity.